### PR TITLE
8284893: Fix typos in java.base

### PR DESCRIPTION
**Pull request by magicus**
I ran `codespell` on the `src/java.base` directory, and accepted those changes where it indeed discovered real typos.

(Due to false positives this can unfortunately not be run automatically) 

The majority of fixes are in comments. A handful is in strings, one in a local variable name, and a couple in parameter declarations.

Annoyingly, there are several instances of "childs" (instead of "children") in the source code, but they were not local and I dared not change them. Someone braver than me might take a stab at it, perhaps..

---------
### Progress
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue
- [x] Change must be properly reviewed

### Issue
 * [JDK-8284893](https://bugs.openjdk.java.net/browse/JDK-8284893): Fix typos in java.base

### Reviewers
 * [Iris Clark](https://openjdk.java.net/census#iris) (@irisclark - **Reviewer**) ⚠️ Review applies to 5e3a01c60fc05b7037093ffe8c9288efee2c6b91
 * [Bradford Wetmore](https://openjdk.java.net/census#wetmore) (@bradfordwetmore - **Reviewer**) ⚠️ Review applies to 5e3a01c60fc05b7037093ffe8c9288efee2c6b91
 * [Lance Andersen](https://openjdk.java.net/census#lancea) (@LanceAndersen - **Reviewer**)
 * [Sean Mullan](https://openjdk.java.net/census#mullan) (@seanjmullan - **Reviewer**)
 * [Naoto Sato](https://openjdk.java.net/census#naoto) (@naotoj - **Reviewer**)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.java.net/jdk pull/8250/head:pull/8250` \
`$ git checkout pull/8250`

Update a local copy of the PR: \
`$ git checkout pull/8250` \
`$ git pull https://git.openjdk.java.net/jdk pull/8250/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 8250`

View PR using the GUI difftool: \
`$ git pr show -t 8250`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.java.net/jdk/pull/8250.diff">https://git.openjdk.java.net/jdk/pull/8250.diff</a>

</details>

### CODE REVIEW

File: src/java.base/share/classes/jdk/internal/icu/impl/NormalizerImpl.java [hunk 1]
```diff
@@ -1999,7 +1999,7 @@ public static String canonicalDecomposeWithSingleQuotation(String string) {
                     back=preBack;
                 }
 
-                // this is where we are right now with all these indicies:
+                // this is where we are right now with all these indices:
```
Comment (naotoj) [inline src/java.base/share/classes/jdk/internal/icu/impl/NormalizerImpl.java:2002]: Although these are actual typos, they come from upstream ICU code. Changing them locally would make merging complicated, so please exclude ICU related changes from the PR. I guess fixes in other 3rd party libraries are in the same boat.

**Review by irisclark — APPROVED** [review]
Looks good.

I usually like GitHub's colorful diffs, but this is one of those rare cases where looking at the single webrev-generated diff file is so much easier.  About the only thing that could improve it would be to reduce the context (i.e. "diff -C").

File: src/java.base/share/classes/sun/security/provider/certpath/AdjacencyList.java [hunk 2]
```diff
@@ -125,7 +125,7 @@ private boolean buildList(List<List<Vertex>> theList, int index,
 
         // Each time this method is called, we're examining a new list
         // from the global list. So, we have to start by getting the list
-        // that contains the set of Vertexes we're considering.
+        // that contains the set of Vertices we're considering.
```
Comment (bradfordwetmore) [inline src/java.base/share/classes/sun/security/provider/certpath/AdjacencyList.java:128]: The class being affected is a Vertex, so either change to vertices, or leave as is...

**Review by bradfordwetmore — APPROVED** [review]
I checked over:

java.base/macosx/classes/apple/security
java.base/share/classes/com/sun/crypto
java.base/share/classes/com/sun/security
java.base/share/classes/java/security
java.base/share/classes/javax/crypto
java.base/share/classes/javax/net
java.base/share/classes/sun/security

The copyright dates need updating.

**Comment by turbanoff** [comment]
Shouldn't copyright year be updated too?

File: src/java.base/share/legal/icu.md [hunk 3]
```diff
@@ -307,7 +307,7 @@ property of their respective owners.
  #  Redistributions of source code must retain the above copyright notice, this
  #  list of conditions and the following disclaimer. Redistributions in binary
  #  form must reproduce the above copyright notice, this list of conditions and
- #  the following disclaimer in the documentation and/or ther materials
+ #  the following disclaimer in the documentation and/or the materials
```
Comment (pavelrappo) [inline src/java.base/share/legal/icu.md:310]: I think it's a mistype of "other", not "the"; look at the similar text below.

File: src/java.base/share/native/libzip/zlib/ChangeLog [hunk 4]
```diff
@@ -96,7 +96,7 @@ Changes in 1.2.7.1 (24 Mar 2013)
 - Fix types in contrib/minizip to match result of get_crc_table()
 - Simplify contrib/vstudio/vc10 with 'd' suffix
 - Add TOP support to win32/Makefile.msc
-- Suport i686 and amd64 assembler builds in CMakeLists.txt
+- Support i686 and amd64 assembler builds in CMakeLists.txt
```
Comment (pavelrappo) [inline src/java.base/share/native/libzip/zlib/ChangeLog:99]: Similarly to Naoto's comment on ICU: shouldn't we leave this as is? If anybody cares enough about this typo, they could file a bug against zlib directly.

File: src/java.base/share/native/libzip/zlib/README [hunk 5]
```diff
@@ -64,7 +64,7 @@ Notes for some targets:
 - zlib doesn't work with gcc 2.6.3 on a DEC 3000/300LX under OSF/1 2.1 it works
   when compiled with cc.
 
-- On Digital Unix 4.0D (formely OSF/1) on AlphaServer, the cc option -std1 is
+- On Digital Unix 4.0D (formerly OSF/1) on AlphaServer, the cc option -std1 is
```
Comment (pavelrappo) [inline src/java.base/share/native/libzip/zlib/README:67]: Same as above.

File: src/java.base/share/legal/icu.md (hunk 3 above)
Comment (pavelrappo) [reply src/java.base/share/legal/icu.md:310]: Should be addressed on the ICU side.

File: src/java.base/windows/native/libnio/ch/wepoll.c [hunk 6]
```diff
@@ -891,7 +891,7 @@ int init(void) {
     /* `InitOnceExecuteOnce()` itself is infallible, and it doesn't set any
      * error code when the once-callback returns FALSE. We return -1 here to
      * indicate that global initialization failed; the failing init function is
-     * resposible for setting `errno` and calling `SetLastError()`. */
+     * responsible for setting `errno` and calling `SetLastError()`. */
```
Comment (AlanBateman) [inline src/java.base/windows/native/libnio/ch/wepoll.c:894]: This is also 3rd party code so should be dropped from this patch.

File: src/java.base/share/native/libzip/zlib/inftrees.h [hunk 7]
```diff
@@ -62,7 +62,7 @@ typedef struct {
 /* Maximum size of the dynamic table.  The maximum number of code structures is
    1444, which is the sum of 852 for literal/length codes and 592 for distance
    codes.  These values were found by exhaustive searches using the program
-   examples/enough.c found in the zlib distribtution.  The arguments to that
+   examples/enough.c found in the zlib distribution.  The arguments to that
```
Comment (AlanBateman) [inline src/java.base/share/native/libzip/zlib/inftrees.h:65]: This is 3rd party code so should be dropped from the patch as we want as few changes to this code has possible.

**Comment by AlanBateman** [comment]
I skimmed through the changes to src/java.base and they look okay except for the changes to 3rd party code that I think should be dropped from this patch.

File: src/java.base/share/classes/sun/security/provider/certpath/AdjacencyList.java (hunk 2 above)
Comment (seanjmullan) [reply src/java.base/share/classes/sun/security/provider/certpath/AdjacencyList.java:128]: Agree, suggest changing it to "vertices".

**Comment by magicus** [comment]
It's a bit annoying that 3rd party code is not more distinctly handled in the JDK source. :( I reverted the code pointed out by reviewers, but then later found the ASM code as well.

If I feel really motivated (or bored) I might try to submit PRs with these fixes upstream. Or not.

Everybody happy now?

**Review by LanceAndersen — APPROVED** [review]

**Review by seanjmullan — APPROVED** [review]

**Review by naotoj — APPROVED** [review]

**Pull request closed without merging**